src/vocutouts/workers.py

```diff
@@ -75,111 +86,87 @@ class TaskTransientError(Exception):
+    """
     # Tell UWS that we have started executing.
     message = CurrentMessage.get_current_message()
     now = datetime.now(tz=timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
     job_started.send(job_id, message.message_id, now)
 
-    # Input parameters to the pipeline will be provided as a Butler repository
-    # containing a dataset of type cutout_positions and a data ID matching the
-    # image to which the cutout stencil is being applied.  Create that input
-    # Butler collection and register the data type.  Also determine the name
-    # of the output collection.
-    uuid = uuid4()
-    positions_collection = f"service/cutouts/positions/{uuid}"
-    output_collection = f"service/cutouts/{uuid}"
-    repository = os.environ["CUTOUT_BUTLER_REPOSITORY"]
-    collection = os.environ["CUTOUT_BUTLER_COLLECTION"]
-    butler = Butler(repository, run=positions_collection)
-
-    # This should only need to be done once, but it does have to be done once.
-    # For now, register the dataset type each time, but this is silly so we
-    # should come up with a better initialization process.  (Maybe do this as
-    # part of vo-cutouts init.)
-    dataset_type = DatasetType(
-        "cutout_positions",
-        dimensions=data_id.keys(),
-        universe=butler.registry.dimensions,
-        storageClass="AstropyQTable",
-    )
-    butler.registry.registerDatasetType(dataset_type)
-
-    # Store the cutout parameters in the input Butler collection.
-    #
-    # Use a conversion ratio of 0.2 arcsec/pixel so that we can use the
-    # current test cutout pipeline.  The eventual cutout pipeline will accept
-    # min/max pairs in ICRS coordinates.
-    if any(math.isinf(v) for v in (ra_min, ra_max, dec_min, dec_max)):
-        raise TaskFatalError("UsageError Unbounded ranges not yet supported")
-    pos = SkyCoord(ra_min, dec_min, unit="deg")
-    xspan = ((ra_max - ra_min) * 3600 / 0.2) * u.dimensionless_unscaled
-    yspan = ((dec_max - dec_min) * 3600 / 0.2) * u.dimensionless_unscaled
-    row = 1 * u.dimensionless_unscaled
-    input_table = QTable(
-        [[row], [pos], [xspan], [yspan]],
-        names=["id", "position", "xspan", "yspan"],
-    )
-    butler.put(input_table, "cutout_positions", **data_id)
-
-    # Perform the cutout.  This currently uses subprocess to call pipetask.
-    # The long-term plan is for there to be an API for running a pipeline so
-    # that any setup overhead can be incurred once at worker start and then
-    # cutouts can be done with a direct Python call.
-    #
-    # This tells Butler to register data types every time, which as above
-    # should not be necessary, but they do have to be registered once.
-    data_query_terms = []
-    for key, value in data_id.items():
-        if isinstance(value, int):
-            data_query_terms.append(f"{key}={value}")
+    # Currently, only a single data ID and a single stencil are supported.
+    assert len(dataset_ids) == 1
+    assert len(stencils) == 1
```

Comment:
Not sure why an assert rather than `TaskFatalError`? (I admit I also stay away from asserts for anything other than tests or mypy hints because they can be disabled.

Reply:
They were assertions because the frontend already does this check so this shouldn't be possible, but I agree that there's no real harm in using `TaskFatalError` here instead.